features/FEATURE_EXPERIMENTAL_API/FEATURE_PSA/TARGET_TFM/include/tfm_ns_mailbox.h

```diff
@@ -85,17 +85,27 @@ bool tfm_ns_mailbox_is_msg_replied(mailbox_msg_handle_t handle);
 int32_t tfm_ns_mailbox_init(struct ns_mailbox_queue_t *queue);
 
 /**
- * \brief Get the handle of the current non-secure task executing mailbox
- *        functionalities
+ * \brief Create a binary semaphore for mailbox
  *
  * \note This function should be implemented according to platform, NS OS
  *       and actual use scenario.
  *       This function can be ignored or return NULL if sleep/wake-up mechanism
  *       is not required in PSA Client API implementation.
  *
- * \return Return the handle of task.
+ * \return Return the handle of semaphore. or NULL if failed
  */
-const void *tfm_ns_mailbox_get_task_handle(void);
+void * tfm_ns_mailbox_hal_create_semaphore(void);
```

Comment:
Is this change present in any public branch of TF-M or change request to TF-M?

Reply:
No

Reply:
How will we ensure when Mbed OS upgrades TF-M next time we have the necessary changes here? We need a patch to https://github.com/ARMmbed/trusted-firmware-m/tree/dev/feature-dualcore to keep changes like these. Could you please raise a PR to https://github.com/ARMmbed/trusted-firmware-m/tree/dev/feature-dualcore with the patches we need on TF-M to support PSoC 64?

Reply:
Jaeden, the plan here is to use our binary to get this out the door. We will work post-release to synchronize what we can. There remains a general problem here of supporting different versions of TFM. In this particular instance, I believe we can revert this change but will not be able to in this timeframe.

Reply:
Reverted.